magefile.go

```diff
@@ -1154,15 +1216,319 @@ func packageAgent(platforms []string, packagingFn func()) {
 			panic(err)
 		}
 	}
+}
 
-	// package agent
-	log.Println("--- Running packaging function")
-	packagingFn()
+// simple struct to deserialize branch information.
+// When we remove snapshot API dependency this can go in the artifact api client code
+type branchInfo struct {
+	Version     string `json:"version"`
+	BuildId     string `json:"build_id"`
+	ManifestUrl string `json:"manifest_url"`
+	SummaryUrl  string `json:"summary_url"`
+}
 
-	log.Println("--- Running post packaging ")
-	mg.Deps(Update)
-	mg.Deps(CrossBuild, CrossBuildGoDaemon)
-	mg.SerialDeps(devtools.Package, TestPackages)
+// FetchLatestAgentCoreStagingDRA is a simple mage target that will retrieve the elastic-agent-core DRA artifacts and
+// place them under build/dra/buildID. It accepts one argument that has to be a release branch present in staging DRA
+func FetchLatestAgentCoreStagingDRA(ctx context.Context, branch string) error {
+
+	branchInfo, err := findLatestBuildForBranch(ctx, baseURLForStagingDRA, branch)
+
+	// Create a dir with the buildID at <root>/build/dra/<buildID>
+	repositoryRoot, err := findRepositoryRoot()
+	if err != nil {
+		return fmt.Errorf("finding repository root: %w", err)
+	}
+	draDownloadDir := filepath.Join(repositoryRoot, "build", "dra")
+	err = os.MkdirAll(draDownloadDir, 0o770)
+
+	if err != nil {
+		return fmt.Errorf("creating %q directory: %w", err)
+	}
+
+	artifacts, err := downloadDRAArtifacts(ctx, branchInfo.ManifestUrl, draDownloadDir, agentCoreProjectName)
+
+	if err == nil {
+		fmt.Println("Downloaded agent core DRAs:")
+		for k, _ := range artifacts {
+			fmt.Println(k)
+		}
+	}
+	return err
+}
+
+// PackageUsingDRA packages elastic-agent for distribution using Daily Released Artifacts specified in manifest.
+func PackageUsingDRA(ctx context.Context) error {
+
+	start := time.Now()
+	defer func() { fmt.Println("package ran for", time.Since(start)) }()
+
+	platforms := devtools.Platforms.Names()
+	if len(platforms) == 0 {
+		return fmt.Errorf("elastic-agent package is expected to build at least one platform package")
+	}
+
+	if !devtools.PackagingFromManifest {
+		return fmt.Errorf("elastic-agent PackageUsingDRA is expected to build from a manifest. Check that %s is set to a manifest URL", devtools.ManifestUrlEnvVar)
+	}
+
+	manifestUrl := devtools.ManifestURL
+
+	build, err := manifest.DownloadManifest(manifestUrl)
+	if err != nil {
+		return fmt.Errorf("downloading manifest from %q: %w", manifestUrl, err)
```

Comment:
the failure is implicit in the fact that it's an error object, the string is only for context. I wanted to avoid the pattern:
```
failed thing A: failed nested operation B : <real error here>
```
if we avoid the `failed` we have
```
thing A: nested operation B: <real error here>
```

Reply:
I thought that was your idea. What I intended to avoid with those suggestions is the situation where the errors end up being printed without further explanation and then it could look like an informational log rather than an error.